lib/pure/asynchttpserver.nim

```diff
@@ -372,19 +372,25 @@ const
     ## This can be set on the command line during compilation
     ## via `-d:nimMaxDescriptorsFallback=N`
 
+proc listen*(server: AsyncHttpServer; socket: AsyncSocket) {.inline.} =
+  # Listen on a provided socket. This is caller's responsibility to
+  # call `register()` before `listen()`
+  server.socket = socket
+  server.socket.listen()
+
 proc listen*(server: AsyncHttpServer; port: Port; address = ""; domain = AF_INET) =
   ## Listen to the given port and address.
   when declared(maxDescriptors):
     server.maxFDs = try: maxDescriptors() except: nimMaxDescriptorsFallback
   else:
     server.maxFDs = nimMaxDescriptorsFallback
-  server.socket = newAsyncSocket(domain)
+  let socket = newAsyncSocket(domain)
```

Comment:
I don't think this will work. The `server.socket` will never be set if this proc is called.

Reply:
You are right.